consensus/ibft/signer/extra.go

```diff
@@ -0,0 +1,250 @@
+package signer
+
+import (
+	"fmt"
+
+	"github.com/0xPolygon/polygon-edge/types"
+	"github.com/0xPolygon/polygon-edge/validators"
+	"github.com/umbracle/fastrlp"
+)
+
+var (
+	// IstanbulDigest represents a hash of "Istanbul practical byzantine fault tolerance"
+	// to identify whether the block is from Istanbul consensus engine
+	IstanbulDigest = types.StringToHash("0x63746963616c2062797a616e74696e65206661756c7420746f6c6572616e6365")
+
+	// IstanbulExtraVanity represents a fixed number of extra-data bytes reserved for proposer vanity
+	IstanbulExtraVanity = 32
+
+	// IstanbulExtraSeal represents the fixed number of extra-data bytes reserved for proposer seal
+	IstanbulExtraSeal = 65
+
+	zeroBytes = make([]byte, 32)
+)
+
+// IstanbulExtra defines the structure of the extra field for Istanbul
+type IstanbulExtra struct {
+	Validators           validators.Validators
+	ProposerSeal         []byte
+	CommittedSeals       Sealer
+	ParentCommittedSeals Sealer
+}
+
+type Sealer interface {
```

Comment:
`Sealer` seems like a misnomer as it doesn't do anything but rlp encoding and `Num()`. `Seals` might be a more appropriate name